libraries/AP_Logger/AP_Logger.cpp

```diff
@@ -36,7 +36,11 @@ extern const AP_HAL::HAL& hal;
 #endif
 
 #ifndef HAL_LOGGING_STACK_SIZE
-#define HAL_LOGGING_STACK_SIZE 1580
+    #if CONFIG_HAL_BOARD == HAL_BOARD_ESP32
```

Comment:
Why not define this in the AP_HAL_Boards.h file - or in the ESP32 include file included from there?

That's why the #ifdef exists here!

Reply:
I was unaware that chibios had n internal "extra bit" that it adds to each thread created through Scheduler::thread_create ... i've done equivalent in the esp32 HAL now.